stdlib/array.ncl

```diff
@@ -280,5 +280,50 @@
           l
         else
           (sort cmp (parts.right)) @ [first] @ (sort cmp (parts.wrong)),
+
+    optionals : forall a. Bool -> Array a -> Array a
+      | doc m%"
+        Takes a Bool and an Array and returns the array back if it's true,
+        otherwise, it returns an empty array.
+
+        For example:
+        ```nickel
+          optionals true [1, 2, 3] => [1, 2, 3]
+          optionals false [`a, `b, `c] => []
+        ```
+      "%
+      = fun condition values => if condition then values else [],
+
+    flat_map : forall a b. (a -> Array b) -> Array a -> Array b
+      | doc m%"
+        The flatmap operation for arrays. First maps the given function over the
+        array and then concatenates the result.
```

Comment:
Or can also keep concatenate only, but then I think it needs to be expanded. Concatenate the result sounds a bit strange, because in general you concatenate many pieces together, not "one thing". 

```suggestion
        array and then flatten the result.
```